### PR TITLE
Port over fix to effects-from-enchantments

### PR DESCRIPTION
**Pull request by chaosvolt**
#### Summary

SUMMARY: Bugfixes "Port over fix for applying effects from passive enchantments"

#### Purpose of change

An important fix to enchantment code that I forgot about. Needed for effects-from-enchantments to function again, especially if from mutations.

#### Describe the solution

1. Ported over changes to item.cpp and suffer.cpp, for less broken processing of passive relic/enchantment effects. This had to be done as `enchantment_cache.activate_passive` instead of `enchantment_cache->activate_passive` but this causes a problem, see below.
2. Altered the stat adjustment of the cube of shame to test an issue that came up while working on initial commit.

#### Describe alternatives you've considered

Continue to slowly go insane.

#### Testing

1. Compiled and load-tested.
2. Spawned in and activated the cube of shame.
3. Tested the Cloaking System CBM to confirm invisibility effect appears on activation.
4. Ported the BN version of Arcana over temporaily to test Draconic Wings, effect shows up on character screen and fall damage negated.
5. Also waited a while with the currently WIP version of the Life Sign Supression CBM active, confirming healthiness stats went down.

#### Additional context

Source PR: https://github.com/CleverRaven/Cataclysm-DDA/pull/43101

Changes to TEST_DATA content not ported over, as that would require me to also port over transforming mutations. Something that'd be useful to port over in the future, but not a vital fix like this is. It will be needed if I want to port over https://github.com/CleverRaven/Cataclysm-DDA/pull/45687 however, which I would like to eventually.

Had an issue with the double-stat bug recurring for reasons I could've figure out, at first. Fix is entirely thanks to Coolthulhu finding the cause after discussion on the modder's discord, I was 100% stumped here and the cause turned out to be in a file I hadn't even thought to scrutinize.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
* Spawn cube of shame
* Wield it
* Note stats didn't change
* Activate it

Expected:
* Stats drop

Got:
* All stats just as before

**Comment by chaosvolt** [comment]
Wait one turn, relic effects have a delay written into them and I'm not sure why they do that. Same issue can be seen in DDA, has been part of relic code since the beginning it seems.

**Comment by Coolthulhu** [comment]
That's with waiting.

**Comment by Coolthulhu** [comment]
I also spawned a fresh cube, to make sure it's not using old relic data.

**Comment by chaosvolt** [comment]
That's...odd. It was working when I did the test right before committing my changes.

**Comment by chaosvolt** [comment]
After double-checking some things it turns out that this bug WAS actually fixed pre-nested, which I could've sworn it was. This PR evidently re-introduces that bug somehow, going to mark as WIP again until I figure out how to fix that one.

File: src/player.cpp [hunk 1]
```diff
@@ -3429,6 +3429,7 @@ void player::use( item_location loc )
     } else {
         add_msg( m_info, _( "You can't do anything interesting with your %s." ),
                  used.tname() );
+        recalculate_enchantment_cache();
     }
```
Comment (olanti-p) [inline src/player.cpp:3433]:
```suggestion
    }
    recalculate_enchantment_cache();
```

**Comment by Coolthulhu** [comment]
Test cases work. Looks good to go.

**Comment by chaosvolt** [comment]
Thank you for the merge, and sorry for the odd problems had during previous tests.

**Pull request merged**